wire-grpc-client/src/main/java/com/squareup/wire/GrpcMethod.kt

```diff
@@ -211,6 +220,154 @@ internal sealed class GrpcMethod<S : Any, R : Any>(
     }
   }
 
+  class GrpcCall<S : Any, R : Any>(
+    path: String,
+    requestAdapter: ProtoAdapter<S>,
+    responseAdapter: ProtoAdapter<R>
+  ) : GrpcMethod<S, R>(path, requestAdapter, responseAdapter) {
+    override fun invoke(
+      grpcClient: GrpcClient,
+      args: Array<Any>
+    ): com.squareup.wire.GrpcCall<S, R> {
+      return object : com.squareup.wire.GrpcCall<S, R> {
+        // TODO(benoit) We cannot eagerly create it because we're waiting on the request.
+        private var call: Call? = null
+
+        // TODO(benoit) How do we deal with `Call` being lazily instantiated?
+        override val timeout: Timeout
+          get() = call!!.timeout()
+
+        override fun cancel() {
+          call?.cancel()
```

Comment:
nit: Add a `checkNotNull()` to verify that we're in a correct state?

Reply:
I’d like to honor cancel whenever, and just store state in this when necessary

Reply:
I don't follow Jesse, is the current impl proper then?